libraries/AP_Compass/CompassCalibrator.cpp

```diff
@@ -79,7 +79,8 @@ extern const AP_HAL::HAL& hal;
 ///////////////////// PUBLIC INTERFACE /////////////////////
 ////////////////////////////////////////////////////////////
 
-CompassCalibrator::CompassCalibrator()
+CompassCalibrator::CompassCalibrator() :
+    _sample_buffer(nullptr)
```

Comment:
For this to be a problem we'd have to have allocated this on the stack, essentially: https://ardupilot.org/dev/docs/style-guide.html#implicit-zeroing-of-memory

Do you still see a problem here?

Reply:
Yes, in my case it was on the stack. Thank you for the reference. Feel free to close PR then and sorry for the noise.